integrationTests/MarkOldPasswordedAccountsInStripeTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use Doctrine\DBAL\Connection;
+use MatchBot\Application\Commands\MarkOldPasswordedAccountsInStripe;
+use MatchBot\IntegrationTests\IntegrationTest;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Stripe\Service\CustomerService;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * This test needs access to the identity DB so can only be run in local dev environment - oustide docker.
+ * Run like so:
+ *
+ *
+ *
+ *    $ export IDENTITY_MYSQL_USER=root
+ *    $ export IDENTITY_MYSQL_PASSWORD=tbgLocal123
+ *    $ export IDENTITY_MYSQL_HOST=127.0.0.1
+ *    $ export IDENTITY_MYSQL_PORT=30051
+ *    $ vendor/bin/phpunit --config=phpunit-integration.xml integrationTests/MarkOldPasswordedAccountsInStripeTest.php
+ *
+ */
+class MarkOldPasswordedAccountsInStripeTest extends IntegrationTest
+{
+    use ProphecyTrait;
+
+    public function testFoo(): void
+    {
+        $identityDBConnection = $this->getServiceByName(MarkOldPasswordedAccountsInStripe::IDENTITY_DBAL_CONNECTION_SERVICE_NAME);
+
+        \assert($identityDBConnection instanceof Connection);
+
+        $stripeClientProphecy = $this->prophesize(\Stripe\StripeClient::class);
+
+        $stripeCustomersProphecy = $this->prophesize(CustomerService::class);
+        $stripeClientProphecy->customers = $stripeCustomersProphecy->reveal();
+
+        $sut = new MarkOldPasswordedAccountsInStripe(
+            $this->getService(\Psr\Log\LoggerInterface::class),
+            $stripeClientProphecy->reveal(),
+            $this->prophesize(Redis::class)->reveal(),
+            $identityDBConnection
+        );
+
+        $lockFactoryProphecy = $this->prophesize(\Symfony\Component\Lock\LockFactory::class);
+        $lockFactoryProphecy->createLock('matchbot:mark-old-passworded-accounts-in-stripe', \Prophecy\Argument::cetera())
+            ->willReturn(new \Symfony\Component\Lock\NoLock());
+
+        $sut->setLockFactory($lockFactoryProphecy->reveal());
+
+        $tester = new CommandTester($sut);
+
+        $stripeCustomersProphecy->update('cus_NHDQHoQP8Abgbp', ['metadata' => ['hasPasswordSince' => '2023-02-01 12:01:26']])
```

Comment:
Noel suggested storing the customer email address in stripe along with the password.